### PR TITLE
[Relay] InferCorrectLayout for strided_slice & min_num_branches option in CombineParallelConv2D

### PR DESCRIPTION
**Pull request by vinx13**
In `AlterOpLayout`, rewrite attributes of strided_slice if packed layout (e.g. NCHW4c) is used. The `begin` and `end`  are divided by the factor of sub-axis.
This allows the expression after `CombineParallelConv2D` transformation to use packed layout.
An option `min_num_branches` is added to `CombineParallelConv2D` to control when to apply the pass according to performance issue in #2827 

cc @masahi @kevinthesun @yzhliu

### CODE REVIEW

File: src/relay/op/tensor/transform.cc [hunk 1]
```diff
@@ -1703,6 +1703,61 @@ bool StridedSliceRel(const Array<Type>& types,
...
+    auto shape = old_in_shapes[0];
+
+    // NOTE: Discard "const" qualifier here.
+    auto *params = const_cast<StridedSliceAttrs*>(attrs.as<StridedSliceAttrs>());
+
+    Array<Integer> new_begin, new_end;
+
+    for (size_t i = 0; i < params->begin.size(); i++) {
+      const LayoutAxis& axis = layout[i];
+      CHECK(axis.IsPrimal());
```
Comment (yzhliu) [inline src/relay/op/tensor/transform.cc]: should we return undef instead of failing?

Comment (vinx13) [reply src/relay/op/tensor/transform.cc]: are there cases where original layout contain splitted axes?

Comment (yzhliu) [reply src/relay/op/tensor/transform.cc]: well, in general original layout could be anything valid. It does not contain split axes in most cases but I think we'd better make it work for general cases if possible. e.g. if someone decided to take an MKLDNN-optimized graph as input somehow, then there will be NCHW16c layouts.

**Comment by yzhliu** [comment]
Thanks! @vinx13

**Pull request merged**